### PR TITLE
Suppress logging for configured exceptions thrown by Amazon Lambda handlers

### LINKED ISSUE #24847: Avoid logging certain exceptions thrown by AWS Lambda handlers
### Description

The AWS Lambda extension currently handles exceptions thrown by `RequestHandler` implementations in a uniform manner by logging the execution failure along with the stacktrace. In some cases, however, raising an exception is the expected way to provide a negative response to the caller - e.g. https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-authentication.html#aws-lambda-triggers-pre-authentication-example. In these cases, it's undesirable to have the exception logged. Suppressing the `AbstractLambdaPollLoop` logger via configuration is an option, of course, but then you have to be sure to catch and log any exceptions you actually want to report in your handler code.

It would be great if the logic in the poll loop could make a determination about whether or not to log the exception in a configurable/programmable way.

### Implementation ideas

My first thought is to introduce a specialized `RuntimeException` type that developers can throw directly, or extend with their own exceptions. This makes the logic in `AbstractLambdaPollLoop` trivial, but would mean writing Quarkus-specific handlers.

An alternative would be to allow the configuration of arbitrary exceptions by class name that the extensions should ignore.

### PR DESCRIPTION
**Pull request by emattheis**
Fixes #24847

### CODE REVIEW

File: extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java [hunk 1]
```diff
@@ -35,6 +35,10 @@ public AbstractLambdaPollLoop(ObjectMapper objectMapper, ObjectReader cognitoIdR
         this.launchMode = launchMode;
     }
 
+    protected boolean shouldLog(Exception e) {
```
Comment (patriot1burke) [inline extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java:38]: Always true, shouldn't this check listed exceptions?

Comment (patriot1burke) [reply extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java:38]: never mind!

**Comment by patriot1burke** [comment]
@emattheis I don't see a test case.

File: extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java (hunk 1 above)
Comment (emattheis) [reply extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java:38]: 😛 Yeah, I wasn't sure if the behavior made sense in the Funqy and REST implementations...

**Comment by emattheis** [comment]
Not sure how to test the desired behavior since the goal is to suppress the exceptions from the log, but I suppose I can make a test case that at least covers the new configuration?

**Comment by patriot1burke** [comment]
> Not sure how to test the desired behavior since the goal is to suppress the exceptions from the log, but I suppose I can make a test case that at least covers the new configuration?

Eh, nvm...didnt' realize it was just log suppression.

**Review by patriot1burke — APPROVED** [review]

**Pull request merged**